packages/tx/src/transaction.ts

```diff
@@ -33,8 +33,11 @@ export default class Transaction {
   public readonly s?: BN
 
   public static fromTxData(txData: TxData, opts?: TxOptions) {
-    const { nonce, gasLimit, gasPrice, to, value, data, v, r, s } = txData
-
+    const { nonce, gasLimit, gasPrice, value, data, v, r, s } = txData
+    let to = txData.to
+    if (to instanceof Address) {
```

Comment:
This is ugly. `TxOptions.to` has type `AddressLike` which is not handy IMO. This is of type `Address | Buffer | string`. In case it is an `Address` we use `.buf` to get the `Buffer`. Any thoughts here?

Reply:
Hmmm since `Address` has `address.toBuffer()` I would suggest we add an additional switch to [toBuffer](https://github.com/ethereumjs/ethereumjs-util/blob/master/src/bytes.ts#L111) with:
`} else if(v.toBuffer) { v = v.toBuffer() }`

Then the code here won't need changing at all, since it is `toBuffer(to)` below.

Reply:
I can open a PR for the above ^

Reply:
I will add this here, should have researched this `Address` thing a bit more. Just console.log'd it and then saw it had a `buf` attribute 😝

Reply:
I tried this, but it seems that this `Address` is different from the `ethereumjs-util` address; it does not have `toBuffer`?

This only happens in the from-rpc test, maybe it has to do how `Object.create` copies data? It could copy the `Address` attributes but not the prototype of it?

Reply:
ohh sorry you are right there is no toBuffer on Address, I was getting confused with a prior implementation.

I might like to add that to the util version then, since it would then conform to our type TranformableToBuffer and be more flexible. It would just return the internal buf as you discovered :)

Reply:
We should be careful by returning buffers and make a choice if we want to return the Buffer itself or a clone of this Buffer. I am more in favor here of copying a Buffer, since `toBuffer` semi-implies that this creates a new Buffer. If we don't do this we can get the same weird effects as with `BN` (the `i*` methods) where you edit the Buffer but this thus has the side effect that it also changes `buf` in the Address.

Reply:
I agree that it should be `TransformableToBuffer`, though 👍 😄

Reply:
sounds good, and good idea I made sure to return a copy. See PR: https://github.com/ethereumjs/ethereumjs-util/pull/277

Reply:
released in ethereumjs-util v7.0.6, will integrate now